### DISCUSSION TITLE
v0.1.0-alpha.8

### DISCUSSION THREAD

**Discussion by gaocegege** [tensorchord/envd#369] [Version releases]
## What's Changed
* docs: Edits on README.md by @terrytangyuan in https://github.com/tensorchord/envd/pull/305
* ci: Enable dependabot for Go modules and GitHub Actions packages by @terrytangyuan in https://github.com/tensorchord/envd/pull/306
* fix: add api/__init__.py by @kemingy in https://github.com/tensorchord/envd/pull/317
* ci: Auto-approve and merge dependabot PRs by @terrytangyuan in https://github.com/tensorchord/envd/pull/315
* chore: Enable CI for PRs by @gaocegege in https://github.com/tensorchord/envd/pull/319
* chore(deps): bump github.com/urfave/cli/v2 from 2.4.0 to 2.8.1 by @dependabot in https://github.com/tensorchord/envd/pull/312
* chore(deps): bump github.com/stretchr/testify from 1.7.0 to 1.7.2 by @dependabot in https://github.com/tensorchord/envd/pull/310
* chore(deps): bump github.com/pkg/sftp from 1.13.4 to 1.13.5 by @dependabot in https://github.com/tensorchord/envd/pull/309
* chore(deps): bump github.com/moby/buildkit from 0.10.1 to 0.10.3 by @dependabot in https://github.com/tensorchord/envd/pull/313
* fix(release): Change docker user by @gaocegege in https://github.com/tensorchord/envd/pull/321
* chore(CI): Enable code coverage by @gaocegege in https://github.com/tensorchord/envd/pull/323
* fix: fix typo by @hezhizhen in https://github.com/tensorchord/envd/pull/324
* fix: typo in readme by @kemingy in https://github.com/tensorchord/envd/pull/325
* fix: pypi sdist by @kemingy in https://github.com/tensorchord/envd/pull/318
* workflow: enable macOS in CI without conditions by @rudeigerc in https://github.com/tensorchord/envd/pull/327
* fix: Disable unit test in macOS by @gaocegege in https://github.com/tensorchord/envd/pull/328
* chore: Add more e2e test cases by @gaocegege in https://github.com/tensorchord/envd/pull/330
* chore(dep): manually bump docker distribution 2.8.1 by @3AceShowHand in https://github.com/tensorchord/envd/pull/333
* feat: Suport conda env by @gaocegege in https://github.com/tensorchord/envd/pull/335
* feat: Support specify number of GPUs by @terrytangyuan in https://github.com/tensorchord/envd/pull/336
* chore: Add contributors section in README by @gaocegege in https://github.com/tensorchord/envd/pull/342
* fix: remove py wrapper traceback information by @kemingy in https://github.com/tensorchord/envd/pull/341
* Example: add cpu example by @VoVAllen in https://github.com/tensorchord/envd/pull/338
* docs: add popfido as a contributor for platform by @allcontributors in https://github.com/tensorchord/envd/pull/345
* docs: add popfido as a contributor for doc by @allcontributors in https://github.com/tensorchord/envd/pull/349
* doc: add conda commands documentation by @popfido in https://github.com/tensorchord/envd/pull/351
* fix: config pip source speed up in china by @kenwoodjw in https://github.com/tensorchord/envd/pull/354
* docs: add kenwoodjw as a contributor for code by @allcontributors in https://github.com/tensorchord/envd/pull/355
* ci: Auto-generate changelog via GitHub Actions by @terrytangyuan in https://github.com/tensorchord/envd/pull/359
* docs: Exclude certain types of commits from changelog by @terrytangyuan in https://github.com/tensorchord/envd/pull/360
* enhancement(CLI): Update the description of envd by @jiayouxujin in https://github.com/tensorchord/envd/pull/364
* docs: add jiayouxujin as a contributor for code by @allcontributors in https://github.com/tensorchord/envd/pull/366

## New Contributors
* @dependabot made their first contribution in https://github.com/tensorchord/envd/pull/312
* @hezhizhen made their first contribution in https://github.com/tensorchord/envd/pull/324
* @3AceShowHand made their first contribution in https://github.com/tensorchord/envd/pull/333
* @allcontributors made their first contribution in https://github.com/tensorchord/envd/pull/345
* @popfido made their first contribution in https://github.com/tensorchord/envd/pull/351
* @kenwoodjw made their first contribution in https://github.com/tensorchord/envd/pull/354
* @jiayouxujin made their first contribution in https://github.com/tensorchord/envd/pull/364

**Full Changelog**: https://github.com/tensorchord/envd/compare/v0.1.0-alpha.6...v0.1.0-alpha.8

<hr /><em>This discussion was created from the release <a href='https://github.com/tensorchord/envd/releases/tag/v0.1.0-alpha.8'>v0.1.0-alpha.8</a>.</em>

**Comment by gaocegege** [comment]
Not sure why the release workflow does not work. I will run it locally.

**Reply by Xuanwo** [reply]
![image](https://user-images.githubusercontent.com/5351546/173989767-25948582-cbba-449e-bd65-e276aa883f2b.png)

~~I believe it's required to set `branches` too.~~ Seems wrong, current behavior should works.

References: https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#example-including-branches-and-tags

**Reply by Xuanwo** [reply]
![image](https://user-images.githubusercontent.com/5351546/173990030-e377dc65-4cc1-456c-b2a6-7edb5a1c5383.png)

Release and Changelog workflow not triggered by tags after v0.1.0-alpha.7

**Reply by gaocegege** [reply]
Yep, v0.1.0-alpha.6 is released before the open source. And after that releases fail.

**Reply by gaocegege** [reply]
I think it is related to forks. The 0.1.0-alpha.8 release workflow runs in my personal fork https://github.com/gaocegege/envd/actions

**Reply by Xuanwo** [reply]
Sounds reasonable. Tags should be added upstream instead of fork.

**Reply by gaocegege** [reply]
https://github.com/argoproj/argo-workflows/blob/master/.github/workflows/release.yaml

But argo-workflows uses the same config as us.

**Reply by Xuanwo** [reply]
I think the point is we should tag against `tensorchord/envd` instead of `gaocegege/envd`.

**Reply by gaocegege** [reply]
Yep, we also have the tag created by GitHub in tensorchord/envd https://github.com/tensorchord/envd/releases/tag/v0.1.0-alpha.8

**Reply by Xuanwo** [reply]
What does it mean of `the tag created by GitHub`? The tag `v0.1.0-alpha.7` is not created by hand?

**Reply by gaocegege** [reply]
Yep, it is created by `Draft a new release` on the release page https://github.com/tensorchord/envd/releases/new

![image](https://user-images.githubusercontent.com/5100735/173992907-6677b706-7d9a-4afd-8513-d0edd8508a75.png)

I also tried to create the tag manually, it also does not work.

**Reply by gaocegege** [reply]
I manually created the tag v0.1.0-alpha.9 on upstream. GitHub sends the push event but not sure why github action is not triggered:

```
{
  "ref": "refs/tags/v0.1.0-alpha.9",
  "repository": {
    "id": 480303698,
    "node_id": "R_kgDOHKDaUg",
    "name": "envd",
    "full_name": "tensorchord/envd",
    "private": false,
    "owner": {
      "name": "tensorchord",
      "email": null,
      "login": "tensorchord",
```

**Reply by Xuanwo** [reply]
Interesting, I'm taking this challenge.

**Reply by gaocegege** [reply]
Thanks! I can give you admin permission temporarily if you need it.

**Reply by Xuanwo** [reply]
I will verify some ideas in my own fork first~

**Reply by Xuanwo** [reply]
Please take https://github.com/Xuanwo/envd/commit/deafe9a331ee1443bb1cea1eb29d361660a4564d a try

**Reply by Xuanwo** [reply]
> Please take [Xuanwo@deafe9a](https://github.com/Xuanwo/envd/commit/deafe9a331ee1443bb1cea1eb29d361660a4564d) a try

Wired, not related to this change.

**Reply by Xuanwo** [reply]
Maybe github just missing some events....

Please try again without changing anything to see if it works.

![image](https://user-images.githubusercontent.com/5351546/173995138-b0734eb4-209f-4fdf-836d-48645524e354.png)

The workflow runs as expected, no matter whether I push tags manually or use github's draft releases.

**Reply by gaocegege** [reply]
It also works on my personal fork. But it does not work on upstream. It's weird..

**Reply by Xuanwo** [reply]
> It also works on my personal fork. But it does not work on upstream. It's weird..

Can you take another try now? My fork can't work at first, but works after seconds. I think we should blame github.

**Reply by Xuanwo** [reply]
Oh, I got another idea: the latest commit is triggered by a bot account, please try push a new commit and then tag it.

![image](https://user-images.githubusercontent.com/5351546/173996340-938ff1a2-cb58-4d8f-857a-f780cc744c01.png)

The commit itself doesn't trigger any actions.

**Reply by gaocegege** [reply]
SGTM, I will give it another try.

**Reply by gaocegege** [reply]
It works!
